evalml/automl/automl_search.py

```diff
@@ -205,22 +205,21 @@ def __init__(self,
         additional_objectives = [self._validate_objective(obj) for obj in additional_objectives]
         self.additional_objectives = additional_objectives
 
-        if max_time is None or isinstance(max_time, (int, float)):
-            self.max_time = max_time
-        elif isinstance(max_time, str):
-            self.max_time = convert_to_seconds(max_time)
-        else:
-            raise TypeError("max_time must be a float, int, or string. Received a {}.".format(type(max_time)))
-
+        if not isinstance(max_time, (int, float, str, type(None))):
+            raise TypeError(f"Parameter max_time must be a float, int, string or None. Received {str(max_time)}.")
+        if isinstance(max_time, (int, float)) and max_time <= 0:
```

Comment:
`max_time == 0` seems to be counterintuitive to me. Not sure if we want to leave that as an option for users, or enforce that it must be strictly larger than 0 when it's provided

Reply:
Oh wait, I misunderstood this. This check looks good